### DISCUSSION TITLE
potential path problem for !dcm2niix

### DISCUSSION THREAD

**Discussion by scott-love** [neurodesk/neurodesk.github.io-hugo#383] [General]
Hi all,

I want to start a pre-processing pipeline by converting dicoms to nii using dcm2niix. I didn't find a specific module for dcm2niix but I found it in mricrogl/1.2.20211006. First question, is that the best way to get access to dcm2niix?

Unfortunately, I am unable to get this dcm2niix to see my dicom files. The errors appear to be "path related" which is why I think it cannot "see" my dicoms. I have tried with full and relative paths to the dicom folder. I have tried making the dicom folder the current working directory and running dcm2niix . but without success. A minimal example can be seen in the screen shot attached. 

I should say that I am having this problem using a local session of Neurodesk (v1.2.1) ; I have tried on both an Intel and M1 based Mac and get the same problem. However, the same code as in the minimal example (except for colab setup) but ran in Google Colab actually works - dcm2niix can see and use the dicom files.

Any help on what I am doing wrong here would be much appreciated!

<img width="1220" alt="Screenshot 2023-11-22 at 16 38 09" src="https://github.com/NeuroDesk/neurodesk.github.io/assets/12942961/60551e07-349a-4e34-bb79-3526ee26b40f">

**Comment by stebo85** [comment]
Dear @scott-love,

We can reproduce your problem on our end and we are investigating. And yes, there are better ways of doing it. 

Thank you for flagging this issue
Will update with a suggested solution, improved documentation and potentially a bugfix :)
Steffen

**Comment by stebo85** [comment]
Dear @scott-love,

The quick fix for now is to add 

```
os.environ['SINGULARITY_BINDPATH']='/neurodesktop-storage'
```

at the beginning of your notebook. This variable should have been set but somehow isn't. I am currently investigating why.

A better container to convert dicom-data is "bidstools" - there we bundled all dicom and bids-related tools for such tasks.

Once the bug is fixed we will add an example notebook that shows the use of this as well.

**Reply by scott-love** [reply]
Thanks @stebo85, the quick fix worked great. I will also check out the bidstools container.

**Comment by stebo85** [comment]
Dear @scott-love,

We just released a new version of Neurodesk (Neurodeskapp 1.3.0, Docker image version vnmd/neurodesktop:2023-11-28) which fixes your problem. I haven't added an example yet for the bidstools container, but I hope you got it work? If you got the conversion to work would you mind sending through what you did, so I can include this in the example?

Thank you
Steffen
